bottlecap/src/metrics/enhanced/lambda.rs

```diff
@@ -8,12 +8,16 @@ use tracing::error;
 
 pub struct Lambda {
     pub aggregator: Arc<Mutex<Aggregator<1024>>>,
+    pub config: Arc<crate::config::Config>,
```

Comment:
do we really need to pass this config around?
It seems it is only used for `self.config.enhanced_metrics`.
I would either just pass that single boolean or, even better, have 2 Lambda Impl:
- EnhancedLambda when the boolean is true
- NoEnahcnedLambda otherwise
so we save all those ifs around the code too

Reply:
if we have no enhanced lambda then we have to duplicate every method in an entire module, that seems overly complicated because we'd need to always ensure both modules have the same methods and are tested.

This way at least it's clear they become no-ops.

We could pass just the boolean though, but it's an ARC for a slice of memory that always lives for the lifetime of our application. Passing the boolean will actually allocate more memory (albeit one additional bit, maybe 2 for the pointer)

Reply:
Good point, I have the feeling there is a way to remove the extra methods/ifs and handle the enhanced metrics, but maybe it will be clearer later on

Reply:
Yeah I agree, but I only had a minute or two to write this code and didn't think of it. Honestly this is pretty pragmatic already, and prevents us from wrapping the rest of the usages with conditionals.